### DISCUSSION TITLE
@testing-library/user-event for testing?

### DISCUSSION THREAD

**Discussion by fadi-george** [react-hook-form/react-hook-form#2834] [Q&A] [answered]
Hello, Is there any way to use testing library user-event for changing a registered input?

**Comment by bluebill1049** [comment]
can you provide more details? maybe a codesandbox

**Comment by fadi-george** [comment]
Ill get on that

**Comment by fadi-george** [comment, marked as answer]
I think user event might work for most cases but will fail for a complicated setup such as in my situation with a controlled material ui timepicker inside a radio option in a radio group.

**Reply by bluebill1049** [reply]
I think you can still get the element and trigger an event tho.

**Reply by fadi-george** [reply]
Also happens w/ textfield inside a label of a checkbox (at least w/ material ui), fireEvent.change seems to work fine no matter how nested it is. Might be a bug w/ user-event library but I haven't had time to setup a sandbox for it.

**Reply by bluebill1049** [reply]
can you try `fireEvent.input`?

**Reply by fadi-george** [reply]
fireEvent.input and fireEvent.change both work. seems like a bug w/ user-event.

**Reply by fadi-george** [reply]
Ill quickly try out set state w/ user event and see

**Reply by fadi-george** [reply]
Yeah, user-event type & textfield inside form-control-label (w/ and w/o hook-form) doesn't work.
